gen3/bin/kube-setup-audit-service.sh

```diff
@@ -55,12 +64,58 @@ EOM
   gen3 secrets sync 'setup audit-g3auto secrets'
 }
 
+setup_sqs() {
+  local sqsName="$(gen3 api safe-name audit-sqs)"
+  local sqsInfo="$(gen3 sqs info $sqsName)"
+  sqsUrl="$(jq -e -r '.["QueueUrl"]' <<< "$sqsInfo")"
+  if [ -n "$sqsUrl" ]; then
+    gen3_log_info "audit-service SQS has already been setup - skipping"
+    return 0
+  fi
+
+  gen3_log_info "setting up audit-service SQS"
+  sqsInfo="$(gen3 sqs create-queue $sqsName)" || exit 1
+  sqsUrl="$(jq -e -r '.["sqs-url"].value' <<< "$sqsInfo")" || { echo "Cannot get 'sqs-url' from output: $sqsInfo"; exit 1; }
+  local sqsSendMessageArn="$(jq -e -r '.["send-message-arn"].value' <<< "$sqsInfo")" || { echo "Cannot get 'send-message-arn' from output: $sqsInfo"; exit 1; }
+  local sqsReceiveMessageArn="$(jq -e -r '.["receive-message-arn"].value' <<< "$sqsInfo")" || { echo "Cannot get 'receive-message-arn' from output: $sqsInfo"; exit 1; }
+
+  local saName
+  local roleName
+
+  # fence can push messages to the queue
```

Comment:
(thinking out loud)

I wonder if this should be in `kube-setup-fence-service.sh`. I could see an argument for having it here, but I also think that the `audit-service` only cares about the receiving policy, and the senders should be responsible for their ability to put to the queue. But then again the queue is created here so 🤷

Maybe it makes sense here since the audit-service "owns" this queue, but I guess I also wonder about the implementation and the order we roll services. Do we have a dependency sequence now that's not obvious? Do we have a way of making sure the apps wait to init until after the queue is created and they're able to put/get from it?

Reply:
Yeah I could move it to `kube-setup-fence-service.sh`. The audit SQS and policies would get created even if the audit-service is not in use though, but that's maybe not a big deal.

> the order we roll services

[right now fence rolls after audit-service](https://github.com/uc-cdis/cloud-automation/blob/aea74471429befbe33aa48d6ee56d054a89fb962/gen3/bin/kube-roll-all.sh#L68-L81)

> Do we have a way of making sure the apps wait to init until after the queue is created and they're able to put/get from it?

the SQS and SQS policies are set up during `kube-setup` and the service container doesn't start until `kube-setup` completes, so i think we should be good?

Reply:
Ah, good point. I didn't think about the fact that setup would always run first anyway. And good point about fence potentially having unused policies!